doc/plugin_tutorial/tuto1/src/g_tuto1.mlg

```diff
@@ -145,10 +145,12 @@ END
   it gives an error message that is basically impossible to understand. *)
 
 VERNAC COMMAND EXTEND ExploreProof CLASSIFIED AS QUERY
-| [ "Cmd9" ] ->
-  { let p = Proof_global.give_me_the_proof () in
-    let sigma, env = Pfedit.get_current_context () in
-    let pprf = Proof.partial_proof p in
-    Feedback.msg_notice
-      (Pp.prlist_with_sep Pp.fnl (Printer.pr_econstr_env env sigma) pprf) }
+| ![ proof ] [ "Cmd9" ] ->
+  { fun ~pstate ->
```

Comment:
Can't we provide a syntax like with attributes such that `![ proof ]` binds `proof` in the body?

Reply:
We could, however it is quite practical to use named arguments here and for example imagine we have:
```ocaml
val foo : pstate:Lemmas.t -> int -> Lemmas.t
```
then you can put in a rule that parses `int_arg`:
```ocaml
  { foo int_arg }
```
by the way it was a large mistake not to use ppx for our preprocessor, it is so annoying not to have syntax support....

Reply:
@ppedrot might not like that partial application style.